src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3574,52 +3574,58 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
         protected BoundExpression BindObjectCreationExpression(ObjectCreationExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            var type = BindType(node.Type, diagnostics).TypeSymbol;
+            var type = BindType(node.Type, diagnostics);
+            var typeSymbol = type.TypeSymbol;
```

Comment:
To minimize the changes from master, consider leaving the original name of the `TypeSymbol`.
```
var typeWithAnnotations = BindType(…);
var type = typeWithAnnotations.TypeSymbol;
```